build_tools/tests/test_memory_monitor.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python3
+"""Tests for memory monitoring functionality."""
+
+import json
+import os
+import subprocess
+import sys
+import tempfile
+import time
+from pathlib import Path
+
+# Add parent directory to path
+sys.path.insert(0, str(Path(__file__).parent.parent))
+
+try:
+    import psutil
+except ImportError:
+    print("WARNING: psutil not installed, skipping memory monitor tests")
+    sys.exit(0)
```

Comment:
https://github.com/ROCm/TheRock/issues/2429 was filed for this broken code pattern